File: Anole/HTMLImageElement.cs
using System;
using System.Drawing;
using System.Xml;
using System.IO;
using System.Reflection;
using System.Windows.Forms;

namespace Anole
{
	/// <summary>
	/// Summary description for HTMLImageElement.
	/// </summary>
	public class HTMLImageElement : HTMLElement
	{
		protected string mySource;
		public string Source
		{
			get{return mySource;}
			set{mySource=value;}
		}

		protected Bitmap myBitmap;
		public Bitmap Bitmap
		{
			get{return myBitmap;}
			set{myBitmap=value;}
		}

		public HTMLImageElement(ref HTMLDocument document, HTMLElement parent, XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);

			//Parse the image name
			mySource=ParseString(xnode,"src","");

			//See if file exists
			if(!System.IO.File.Exists(mySource))
			{
				//See if the file exists in the current directory
				string sPath=Path.GetDirectoryName(Assembly.GetExecutingAssembly().GetModules()[0].FullyQualifiedName) + "\\";
				if(System.IO.File.Exists(sPath+mySource))
				{
					mySource=sPath+mySource;
				}
			}
			if(!System.IO.File.Exists(mySource))
				return;

			//Load the bitmap
			try
			{
				myBitmap=new Bitmap(mySource);
				this.Width=myBitmap.Width;
				this.Height=myBitmap.Height;
			}
			catch(Exception ex)
			{
				MessageBox.Show(ex.Message);
				myBitmap=null;
			}
		}

		public override void Layout(HTMLRenderer renderer, Graphics g)
		{
			if(myBitmap==null) return;
			HTMLRenderBand band=renderer.CurrentBand();
			if(band.WidthRemaining()<myBitmap.Width)
			{
				band=renderer.NewBand();
			}

			band.AddItem(new HTMLRenderImage(this));
			
		}
	}
}

File: Anole/HTMLImageElement.vb
Imports System.IO
Imports System.Reflection

Namespace Anole

    Public Class HTMLImageElement : Inherits HTMLElement

        Protected mySource As String

        Public Property Source() As String
            Get
                Return mySource
            End Get
            Set(ByVal value As String)
                mySource = value
            End Set
        End Property

        Protected myBitmap As Bitmap

        Public Property Bitmap() As Bitmap
            Get
                Return myBitmap
            End Get
            Set(ByVal value As Bitmap)
                myBitmap = value
            End Set
        End Property

        Public Sub New(ByRef document As HTMLDocument, ByVal parent As HTMLElement, ByVal xnode As XmlNode)
            Initialize(document, parent, xnode)

            'Parse the image name
            mySource = ParseString(xnode, "src", "")

            'See if file exists
            If Not File.Exists(mySource) Then
                'See if the file exists in the current directory
                Dim sPath As String = Path.GetDirectoryName(Assembly.GetExecutingAssembly().GetModules()(0).FullyQualifiedName) + "\"
                If File.Exists(sPath + mySource) Then mySource = sPath + mySource
            End If

            If Not File.Exists(mySource) Then Return

            'Load the bitmap
            Try
                myBitmap = New Bitmap(mySource)
                Me.Width = myBitmap.Width
                Me.Height = myBitmap.Height
            Catch ex As Exception
                MessageBox.Show(ex.Message)
                myBitmap = Nothing
            End Try
        End Sub

        Public Overloads Overrides Sub Layout(ByVal renderer As HTMLRenderer, ByVal g As Graphics)
            If myBitmap Is Nothing Then Return

            Dim band As HTMLRenderBand = renderer.CurrentBand()

            If band.WidthRemaining() < myBitmap.Width Then band = renderer.NewBand()
            band.AddItem(New HTMLRenderImage(Me))
        End Sub

    End Class

End Namespace
